src/main/java/teammates/ui/webapi/UpdateFeedbackSessionLogsAction.java

```diff
@@ -35,9 +35,13 @@ public JsonResult execute() {
                 startTime.toEpochMilli(), endTime.toEpochMilli(), null);
 
         Map<UUID, Map<String, Map<UUID, Map<String, Long>>>> lastSavedTimestamps = new HashMap<>();
+        Map<String, Boolean> isCourseMigratedMap = new HashMap<>();
         for (FeedbackSessionLogEntry logEntry : logEntries) {
 
-            if (!isCourseMigrated(logEntry.getCourseId())) {
+            isCourseMigratedMap.computeIfAbsent(logEntry.getCourseId(),
+                    k -> logic.getCourse(logEntry.getCourseId()) == null);
```

Comment:
actually for this, if the course is migrated, it might still exist in datastore, we should be using the isMigrated flag as a check as well (a course is considered migrated if it either does not exist in datastore, or isMigrated is true)